cache-tree.h

```diff
@@ -62,9 +62,11 @@ static inline int write_cache_as_tree(struct object_id *oid, int flags, const ch
 
 static inline int update_main_cache_tree(int flags)
 {
+	struct repository *r = the_repository;
+
 	if (!the_index.cache_tree)
 		the_index.cache_tree = cache_tree();
-	return cache_tree_update(&the_index, flags);
+	return cache_tree_update(r, r->index, flags);
```

Comment:
This might need a bit of explanation... a casual reviewer might point out (and I almost did!) that `r->index` is easily deduced from `r`, so why not change the signature of `cache_tree_update()` to take a `struct repository*` _instead of_ a `struct index_state *`?

The solution to that riddle is, of course, that you can have _temporary_ indexes. Therefore, you might want to update the cache tree for an index _other_ than `r->index`.

Most likely, information like this would be most welcome in the commit message.

Speaking of the commit message: maybe you want to imitate, say, 3a7a698e93e1031c322ab20c0e336e205514c058's commit message? It talks a bit about what it does, but also a bit about "why?", and a bit about "what about ...?".

Reply:
Sure I'll update this with a proper commit message with that explanation :)